.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,29 @@
+name: Build and push Docker image
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
```

Comment:
Nice work on including `workflow_dispatch`, so this Docker build+publish action can be [performed manually within GitHub's UI](https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/) 👍